src/Illuminate/Queue/RedisQueue.php

```diff
@@ -367,7 +375,24 @@ public function getQueue($queue)
      */
     public function getConnection()
     {
-        return $this->redis->connection($this->connection);
+        if (isset($this->redisConnection)) {
+            return $this->redisConnection;
+        }
+
+        $this->redisConnection = $this->redis->connection($this->connection);
+
+        if ($this->redisConnection instanceof PhpRedisConnection) {
+            $this->redisConnection = $this->redis->resolve($this->connection);
+
+            if (defined('\Redis::OPT_COMPRESSION') && $this->redisConnection->client()->getOption(\Redis::OPT_COMPRESSION) !== \Redis::COMPRESSION_NONE) {
+                $this->redisConnection->client()->setOption(\Redis::OPT_COMPRESSION, \Redis::COMPRESSION_NONE);
```

Comment:
Thanks for your input. Since I will have to change that code, I will wait for Taylors answer. 
Maybe you can give me/us a hint why the MULTI (transaction) would be better than the currently used Lua script.
Thanks again!

Reply:
Because `multi()` works with serializer and compression.